_posts/2022-09-08-sprint-planning-velocity-accuracy.md

```diff
@@ -0,0 +1,34 @@
+---
+layout:     post
+title:      "Sprint Planning: Velocity vs Accuracy"
+date:       2022-09-08 12:00:00 +0100
+categories: scm
+---
+
+# Introduction
+
+My team is working using an agile methodology and several weeks ago we were planning the next sprint. I was in a meeting with my PO (Product Owner) to check the last details when I had a very interesting conversation. We were dividing the work for an epic in tasks and estimating the time to complete them. My initial idea was to create the tasks so that they took less than five story points to complete. I think that I got that correctly as the biggest task was estimated to be three story points. That's when my PO came into play and advised me to further subdivide the work. I should create subtasks for the tasks, and estimate them in hours, rather than story points. Moreover, I should plan the subtasks to take less than a day. He wanted the team to be as accurate as possible with the estimates.
+
+# Velocity and Accuracy
+
+I hesitated because that meant more work in the short term, and the long-term benefit was not clear to me. I would have to spend more time on this task in the coming days to further investigate how to implement each of these tasks. So, my PO and I had a long talk about why it was better to increase the estimation accuracy.
+
+This brings to the two terms that I wanted to discuss in this post:
+
+## Velocity
+
+The Oxford dictionary defines it as "*the speed of something in a particular direction*". Increasing velocity means that a given work takes less time to complete.
+
+## Accuracy
+
+The Oxford dictionary defines it as "*the state of being exact or correct*". Increasing the estimation accuracy of a given work means that you can plan those tasks in advance with more certainty that they will be completed on time.
+
+# Velocity vs Accuracy
+
+In my opinion, in an agile environment those two terms are in conflict. As increasing one of them means that the other will be decreased. As an example, to increase the estimation accuracy I had to decrease my velocity. Why? Because I had to investigate more deeply how to implement the epic I postponed doing other work, thus lowering my velocity. In addition, some of this research would not provide any other long-term benefit, as the actual implementation would occur several weeks later and by then I would have forgotten the learnings from the initial research, thus having to re-investigate them again.
+
+On the other hand, making a perfect estimation planning would allow me, and my team, to know in advance how much work I would be able to deliver in a sprint. But that's difficult to achieve (if not impossible), and it means that once I commit to a given amount of time I risk not delivering it on time. This would not be excessively tragic if the deviation in the estimate is low, but it could also be that the deviation is high and this could have big consequences.

```

Comment:
If I may copy you ;)

The Oxford Dictionary defines: 
- _perfect_ as "having everything that is necessary;...";
- _estimate_ as "to form an idea of the cost, size, value etc. of something, but without calculating it exactly";

With that in mind, I partially agree with your statement. We will miss the estimate, it is expected! We want to have minor deviations as possible, that's our goal!

Plans change, and, if we "estimate" less than a day per task, we can track the deviations on a daily basis, so we are able to set/adjust expectations with stakeholders, avoiding surprises by the end of the quarter, or creating action items like "We need help with Feature A, we are behind the schedule there, can someone jump in?".

When we go to the retrospective, those things should appear too, i.e. "I underestimated task **X**, because of **Y**, lack of **W** knowledge, I wasn't counting with **Z** either". Based on that, we create a _lesson's learned_ doc/checklist to avoid that mistake in the future. Again, the better we get at a subject, the better we get at dealing with it.